src/EnergyPlus/Fans.cc

```diff
@@ -2642,8 +2642,7 @@ void FanSystem::set_size(EnergyPlusData &state)
             }
         }
     }
-    Real64 _rhoAir = Psychrometrics::PsyRhoAirFnPbTdbW(state, state.dataLoopNodes->Node(inletNodeNum).Press, inletAirTemp, inletAirHumRat);
-    designPointFEI = report_fei(state, maxAirFlowRate, designElecPower, deltaPress, _rhoAir);
+    designPointFEI = report_fei(state, maxAirFlowRate, designElecPower, deltaPress, state.dataEnvrn->StdRhoAir);
```

Comment:
In the rho_air calculation as it stood, all of the input conditions were the conditions as set at the inlet node of the fan (data members of the class set during sizing). Problem is, the node hasn't been set up when this line is calculated! It's holding its default as-constructed values, which means InletAirTemp = InletAitHumRat = Inlet pressure = 0.0.

I imagined that using StdRhoAir was appropriate only because that's how it's done in [this line of Fans.cc](https://github.com/NREL/EnergyPlus/blob/4d39dff4a6e899d6215b432714ee6cb6f6dad187/src/EnergyPlus/Fans.cc#L1558). However, from your discovery, it appears we have a real bug in both the calculation, and where it's done in the code.

To the other point, one might hope that unit tests would be monolithic. But [this CI result ](https://myoldmopar.github.io/EnergyPlusBuildResults/EnergyPlus-012b2b10c01cbde63bb978441b6396263e969916-x86_64-Linux-Ubuntu-22.04-gcc-11.4-UnitTestsCoverage-Debug.html)says otherwise. Here, it's because `SetOANodeValues `is called from `GetVRFInput`, and `SetOANodeValues `can only depend on the global environment. This kind of unintentional coupling is irrationally hard to expect - but also ubiquitous!

Reply:
> However, from your discovery, it appears we have a real bug in both the calculation, and where it's done in the code.

Could you clarify if this means more changes are coming for this PR?

> This kind of unintentional coupling is irrationally hard to expect - but also ubiquitous!

Yes.  But also, you should have seen the state of unit testing EnergyPlus when we first added it 10 years ago :) :)

Reply:
I think I'd like some input from @nealkruis and @rraustad , given that a) the FEI calculation uses StdRhoAir as an input elsewhere and b) FEI for the node was clearly dead wrong before. My naïve vote is to leave it, and start an Issue about this using Rich's comment as a starting point.

Reply:
The larger problem with FEI is that it is not a single rating metric. i.e., there are not a set of standard conditions it should be calculated at. Instead it can be calculated at any set of operating conditions (called a "duty point"). EnergyPlus is calculating FEI as as a single metric at design flow and, as such, it doesn't seem appropriate to use a different air density. If we were calculating FEI on an hourly basis we would want to use the actual inlet air density, but as a single data point, I think we should either use standard air density or air density based on the elevation of the site.

Reply:
@rraustad Thoughts? It appears that, as a single value, the FEI calculated here will be / has been incorrect in one of multiple possible ways. How can we move forward so this PR isn't responsible for an entirely new FEI model?

Reply:
My recommendation is to revert the change to the FEI calculation and open a new issue specifically for that. The ANSI Standard FEI calculation is using a ratio of moist air density to standard air density = 1.2, which is dry air density at sea level at 21 C. StdRhoAir is used as the denominator in report_fei and is dry air density at elevation. So the FEI equation is using the wrong denominator.

    state.dataEnvrn->StdBaroPress = DataEnvironment::StdPressureSeaLevel * std::pow(1.0 - 2.25577e-05 * state.dataEnvrn->Elevation, 5.2559);
    state.dataEnvrn->StdRhoAir = Psychrometrics::PsyRhoAirFnPbTdbW(
        state, state.dataEnvrn->StdBaroPress, DataPrecisionGlobals::constant_twenty, DataPrecisionGlobals::constant_zero);

This would need to be discussed by the team. but we could fix the denominator in report_fei and then pass in StdRhoAir (StdRhoAir/StdRhoAirAtSeaLevel) or choose a moist air density at elevation, like 24 C dry-bulb and 50% RH, and document and use that ratio (MoistRhoAir/StdRhoAirAtSeaLevel) for the calculation of FEI. Then just document the changes.